# Integrate Europe PMC and arXiv data

## 1. Introduction

The data for the 11,397 PubMed Central articles extracted using the Europe PMC API will be combined with the 118 exceptions and 301 arXiv articles extracted by GROBID, and cleaned to remove missing values and duplicates.



## 2. Import libraries

In [ ]:
import pandas as pd
import pickle

## 3. Import data for 11,397 PMC articles

Load the data for the 11,397 articles with full text column added after using Beautiful Soup to remove tags and strip markup. This includes the rows for the 118 exceptions without full text.

In [ ]:
with open('2023-01-06_search_results_full_text.pickle', 'rb') as f:
  search_results_full_text = pickle.load(f)

In [ ]:
search_results_full_text

pmcid   published     revised  \
0      PMC9549161  2022-09-26  2022-10-14   
1      PMC9539342  2022-09-22  2022-11-12   
2      PMC9357751  2022-12-01  2022-12-05   
3      PMC9346052  2022-08-03  2022-09-05   
4      PMC9775208  2022-12-15  2022-12-25   
...           ...         ...         ...   
11392  PMC6328940  2019-01-01  2020-03-09   
11393  PMC6598402  2019-06-21  2020-09-28   
11394  PMC6481739  2019-02-05  2020-09-28   
11395  PMC6624471  2019-07-05  2020-09-28   
11396  PMC6409730  2019-02-06  2020-09-30   

                                                   title  \
0           Drug repositioning: A bibliometric analysis.   
1      A review on computer-aided chemogenomics and d...   
2      Repurposing Molnupiravir as a new opportunity ...   
3      Scope of repurposed drugs against the potentia...   
4      Drug Repurposing Using Gene Co-Expression and ...   
...                                                  ...   
11392  β-RA reduces DMQ/CoQ ratio and rescues the enc...   
11393  Alzheimer Disease Pathogenesis: Insights From ...   
11394  Modeling cardiac complexity: Advancements in m...   
11395  Tissue Response to Neural Implants: The Use of...   
11396  24th International Symposium on Infections in ...   

                                                 journal  \
0                              Frontiers in pharmacology   
1                         Chemical biology & drug design   
2      Journal of Generic Medicines : Duplicate, mark...   
3                                   Structural chemistry   
4                                                Biology   
...                                                  ...   
11392                            EMBO molecular medicine   
11393                          Frontiers in neuroscience   
11394                                 APL bioengineering   
11395                          Frontiers in neuroscience   
11396              Medical sciences (Basel, Switzerland)   

                                                 authors  \
0      Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   
1      Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   
2                                                      0   
3      Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   
4                                    Mailem RC, Tayo LL.   
...                                                  ...   
11392  Hidalgo-Gutiérrez A, Barriocanal-Casado E, Bak...   
11393                                Chen XQ, Mobley WC.   
11394  Callaghan NI, Hadipour-Lakmehsari S, Lee SH, G...   
11395       Gulino M, Kim D, Pané S, Santos SD, Pêgo AP.   
11396  Artigas A, Carlet J, Garnacho J, Niederman M, ...   

                              doi  \
0       10.3389/fphar.2022.974849   
1              10.1111/cbdd.14136   
2                               0   
3      10.1007/s11224-022-02020-z   
4         10.3390/biology11121827   
...                           ...   
11392     10.15252/emmm.201809466   
11393    10.3389/fnins.2019.00659   
11394           10.1063/1.5055873   
11395    10.3389/fnins.2019.00689   
11396       10.3390/medsci7020023   

                                                 pdf_url  \
0      https://europepmc.org/articles/PMC9549161?pdf=...   
1      https://europepmc.org/articles/PMC9539342?pdf=...   
2      https://europepmc.org/articles/PMC9357751?pdf=...   
3      https://europepmc.org/articles/PMC9346052?pdf=...   
4      https://europepmc.org/articles/PMC9775208?pdf=...   
...                                                  ...   
11392  https://europepmc.org/articles/PMC6328940?pdf=...   
11393  https://europepmc.org/articles/PMC6598402?pdf=...   
11394  https://europepmc.org/articles/PMC6481739?pdf=...   
11395  https://europepmc.org/articles/PMC6624471?pdf=...   
11396  https://europepmc.org/articles/PMC6409730?pdf=...   

                                                    text  
0      Sir James Black, a winner of the 1988 Nobel Pr...  
1      Tight and selecti

Check for missing text by searching for empty strings.

In [ ]:
search_results_full_text[search_results_full_text['text']=='']

pmcid   published     revised  \
1772  PMC9364697  2022-07-06  2022-10-06   

                                                  title  \
1772  3<sup>a</sup> edizione Giornate della ricerca ...   

                                         journal authors  \
1772  Journal of preventive medicine and hygiene       0   

                                     doi  \
1772  10.15167/2421-4248/jpmh2022.63.1s1   

                                                pdf_url text  
1772  https://europepmc.org/articles/PMC9364697?pdf=...

There is an empty string in the text column for 	PMC9364697 (an Italian edition) as GROBID currently only supports English.

We can also see the 118 missing values using the .isna() method.

In [ ]:
search_results_full_text[search_results_full_text['text'].isna()]

pmcid   published     revised  \
64     PMC9538661  2022-10-13  2022-11-22   
188    PMC9794394  2022-12-28  2023-01-02   
212    PMC9538837  2022-10-10  2022-11-22   
256    PMC9788990  2022-12-24  2023-01-02   
275    PMC9794516  2022-12-28  2023-01-02   
...           ...         ...         ...   
9797   PMC7162151  2020-02-28  2020-04-21   
9809   PMC9094125  2022-05-11  2022-07-16   
10399  PMC7492056  2020-09-15  2020-09-28   
10767  PMC8459260  2021-02-25  2022-01-25   
11329  PMC7163523  2019-08-01  2021-02-16   

                                                   title  \
64     Recent Drug Development and Medicinal Chemistr...   
188    Nonstructural protein 1 (nsp1) widespread RNA ...   
212    The Efficacy of Traditional Medicinal Plants i...   
256    Sleep and circadian rhythm disruption alters t...   
275    MultiOMICs landscape of SARS-CoV-2-induced hos...   
...                                                  ...   
9797                                               News.   
9809   Recent advances in metal-organic framework-bas...   
10399                                    Full Issue PDF.   
10767  Getting in touch with your senses: Mechanisms ...   
11329  Literature review of baseline information on n...   

                            journal  \
64                      ChemMedChem   
188                        iScience   
212        Chemistry & biodiversity   
256                        iScience   
275                        iScience   
...                             ...   
9797           Chemistry & industry   
9809                  Nano research   
10399          JACC. CardioOncology   
10767   WIREs mechanisms of disease   
11329  EFSA Supporting Publications   

                                                 authors  \
64         Ghosh AK, Mishevich JL, Mesecar A, Mitsuya H.   
188                       Bermudez Y, Miles J, Muller M.   
212                      Choe J, Har Yong P, Xiang Ng Z.   
256    Taylor L, Von Lendenfeld F, Ashton A, Sanghani...   
275    Pinto SM, Subbannayya Y, Kim H, Hagen L, Górna...   
...                                                  ...   
9797                                                   0   
9809             Yang M, Zhang J, Wei Y, Zhang J, Tao C.   
10399                                                  0   
10767                                Gupta S, Butler SJ.   
11329  Dávalos A, Henriques R, Latasa M, Laparra M, C...   

                                 doi  \
64            10.1002/cmdc.202200440   
188       10.1016/j.isci.2022.105887   
212           10.1002/cbdv.202200655   
256       10.1016/j.isci.2022.105877   
275       10.1016/j.isci.2022.105895   
...                              ...   
9797            10.1002/cind.842_3.x   
9809       10.1007/s12274-022-4302-x   
10399  10.1016/s2666-0873(20)30180-0   
10767              10.1002/wsbm.1520   
11329                              0   

                                                 pdf_url  text  
64     https://europepmc.org/articles/PMC9538661?pdf=...  None  
188    https://europepmc.org/articles/PMC9794394?pdf=...  None  
212    https://europepmc.org/articles/PMC9538837?pdf=...  None  
256    https://europepmc.org/articles/PMC9788990?pdf=...  None  
275    https://europepmc.org/articles/PMC9794516?pdf=...  None  
...                                                  ...   ...  
9797   https://europepmc.org/articles/PMC7162151?pdf=...  None  
9809   https://europepmc.org/articles/PMC9094125?pdf=...  None  
10399  https://europepmc.org/articles/PMC7492056?pdf=...  None  
10767  https://europepmc.org/articles/PMC8459260?pdf=...  None  
11329  https://europepmc.org/articles/PMC7163523?pdf=...  None  

[118 rows x 9 columns]

And the .info() method.

In [ ]:
search_results_full_text.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11397 entries, 0 to 11396
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   pmcid      11397 non-null  object
 1   published  11397 non-null  object
 2   revised    11397 non-null  object
 3   title      11397 non-null  object
 4   journal    11397 non-null  object
 5   authors    11397 non-null  object
 6   doi        11397 non-null  object
 7   pdf_url    11397 non-null  object
 8   text       11279 non-null  object
dtypes: object(9)
memory usage: 801.5+ KB


## 4. Import data for 118 PMC exceptions

Load the data for the 118 articles with full text column added after using GROBID to extract  the text, and Beautiful Soup to remove tags and strip markup.

In [ ]:
with open('2023-01-06_pmc_search_results_full_text_grobid_v2.pickle', 'rb') as f:
  pmc_search_results_full_text_grobid = pickle.load(f)

In [ ]:
pmc_search_results_full_text_grobid

pmcid   published     revised  \
64     PMC9538661  2022-10-13  2022-11-22   
188    PMC9794394  2022-12-28  2023-01-02   
212    PMC9538837  2022-10-10  2022-11-22   
256    PMC9788990  2022-12-24  2023-01-02   
275    PMC9794516  2022-12-28  2023-01-02   
...           ...         ...         ...   
9797   PMC7162151  2020-02-28  2020-04-21   
9809   PMC9094125  2022-05-11  2022-07-16   
10399  PMC7492056  2020-09-15  2020-09-28   
10767  PMC8459260  2021-02-25  2022-01-25   
11329  PMC7163523  2019-08-01  2021-02-16   

                                                   title  \
64     Recent Drug Development and Medicinal Chemistr...   
188    Nonstructural protein 1 (nsp1) widespread RNA ...   
212    The Efficacy of Traditional Medicinal Plants i...   
256    Sleep and circadian rhythm disruption alters t...   
275    MultiOMICs landscape of SARS-CoV-2-induced hos...   
...                                                  ...   
9797                                               News.   
9809   Recent advances in metal-organic framework-bas...   
10399                                    Full Issue PDF.   
10767  Getting in touch with your senses: Mechanisms ...   
11329  Literature review of baseline information on n...   

                            journal  \
64                      ChemMedChem   
188                        iScience   
212        Chemistry & biodiversity   
256                        iScience   
275                        iScience   
...                             ...   
9797           Chemistry & industry   
9809                  Nano research   
10399          JACC. CardioOncology   
10767   WIREs mechanisms of disease   
11329  EFSA Supporting Publications   

                                                 authors  \
64         Ghosh AK, Mishevich JL, Mesecar A, Mitsuya H.   
188                       Bermudez Y, Miles J, Muller M.   
212                      Choe J, Har Yong P, Xiang Ng Z.   
256    Taylor L, Von Lendenfeld F, Ashton A, Sanghani...   
275    Pinto SM, Subbannayya Y, Kim H, Hagen L, Górna...   
...                                                  ...   
9797                                                   0   
9809             Yang M, Zhang J, Wei Y, Zhang J, Tao C.   
10399                                                  0   
10767                                Gupta S, Butler SJ.   
11329  Dávalos A, Henriques R, Latasa M, Laparra M, C...   

                                 doi  \
64            10.1002/cmdc.202200440   
188       10.1016/j.isci.2022.105887   
212           10.1002/cbdv.202200655   
256       10.1016/j.isci.2022.105877   
275       10.1016/j.isci.2022.105895   
...                              ...   
9797            10.1002/cind.842_3.x   
9809       10.1007/s12274-022-4302-x   
10399  10.1016/s2666-0873(20)30180-0   
10767              10.1002/wsbm.1520   
11329                              0   

                                                 pdf_url  \
64     https://europepmc.org/articles/PMC9538661?pdf=...   
188    https://europepmc.org/articles/PMC9794394?pdf=...   
212    https://europepmc.org/articles/PMC9538837?pdf=...   
256    https://europepmc.org/articles/PMC9788990?pdf=...   
275    https://europepmc.org/articles/PMC9794516?pdf=...   
...                                                  ...   
9797   https://europepmc.org/articles/PMC7162151?pdf=...   
9809   https://europepmc.org/articles/PMC9094125?pdf=...   
10399  https://europepmc.org/articles/PMC7492056?pdf=...   
10767  https://europepmc.org/articles/PMC8459260?pdf=...   
11329  https://europepmc.org/articles/PMC7163523?pdf=...   

                                                    text  
64     \nSevere acute respiratory syndrome coronaviru...  
188    \nThe past 20 years have seen the emergence of...  
212    \nCoronavirus disease (Covid- 19) is a human r...  
256    \n\n\nJ o u r n a l P r e -p r o o f\nRespirat...  
275    \nThe rapid emergence of the COVID-19 pandemic...  
... 

There are now no missing values for the text column.

In [ ]:
pmc_search_results_full_text_grobid.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 118 entries, 64 to 11329
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   pmcid      118 non-null    object
 1   published  118 non-null    object
 2   revised    118 non-null    object
 3   title      118 non-null    object
 4   journal    118 non-null    object
 5   authors    118 non-null    object
 6   doi        118 non-null    object
 7   pdf_url    118 non-null    object
 8   text       118 non-null    object
dtypes: object(9)
memory usage: 9.2+ KB


View a few examples of rows from the 118 articles extracted by GROBID to check that they now have full text.

**PMC9538661**

In [ ]:
pmc_search_results_full_text_grobid[pmc_search_results_full_text_grobid['pmcid']=='PMC9538661'].iloc[0]

pmcid                                               PMC9538661
published                                           2022-10-13
revised                                             2022-11-22
title        Recent Drug Development and Medicinal Chemistr...
journal                                            ChemMedChem
authors          Ghosh AK, Mishevich JL, Mesecar A, Mitsuya H.
doi                                     10.1002/cmdc.202200440
pdf_url      https://europepmc.org/articles/PMC9538661?pdf=...
text         \nSevere acute respiratory syndrome coronaviru...
Name: 64, dtype: object

**PMC7163523**

In [ ]:
pmc_search_results_full_text_grobid[pmc_search_results_full_text_grobid['pmcid']=='PMC7163523'].iloc[0]

pmcid                                               PMC7163523
published                                           2019-08-01
revised                                             2021-02-16
title        Literature review of baseline information on n...
journal                           EFSA Supporting Publications
authors      Dávalos A, Henriques R, Latasa M, Laparra M, C...
doi                                                          0
pdf_url      https://europepmc.org/articles/PMC7163523?pdf=...
text         \nThis part provides baseline information on t...
Name: 11329, dtype: object

**PMC7640961**

In [ ]:
pmc_search_results_full_text_grobid[pmc_search_results_full_text_grobid['pmcid']=='PMC7640961'].iloc[0]

pmcid                                               PMC7640961
published                                           2020-09-27
revised                                             2020-12-18
title        Repurposing of FDA-Approved Toremifene to Trea...
journal                           Journal of proteome research
authors                                    Martin WR, Cheng F.
doi                              10.1021/acs.jproteome.0c00397
pdf_url      https://europepmc.org/articles/PMC7640961?pdf=...
text         \nAs of August 4, 2020, there are over 18 mill...
Name: 2583, dtype: object

## 5. Concatenate PMC DataFrames

Concatenate the two PMC DataFrames and then drop the duplicates keeping the 118 rows with text.

In [ ]:
pmc_search_results_full_text_merged = pd.concat([search_results_full_text, pmc_search_results_full_text_grobid]).drop_duplicates(subset=['pmcid'], keep='last')

In [ ]:
pmc_search_results_full_text_merged

pmcid   published     revised  \
0      PMC9549161  2022-09-26  2022-10-14   
1      PMC9539342  2022-09-22  2022-11-12   
2      PMC9357751  2022-12-01  2022-12-05   
3      PMC9346052  2022-08-03  2022-09-05   
4      PMC9775208  2022-12-15  2022-12-25   
...           ...         ...         ...   
9797   PMC7162151  2020-02-28  2020-04-21   
9809   PMC9094125  2022-05-11  2022-07-16   
10399  PMC7492056  2020-09-15  2020-09-28   
10767  PMC8459260  2021-02-25  2022-01-25   
11329  PMC7163523  2019-08-01  2021-02-16   

                                                   title  \
0           Drug repositioning: A bibliometric analysis.   
1      A review on computer-aided chemogenomics and d...   
2      Repurposing Molnupiravir as a new opportunity ...   
3      Scope of repurposed drugs against the potentia...   
4      Drug Repurposing Using Gene Co-Expression and ...   
...                                                  ...   
9797                                               News.   
9809   Recent advances in metal-organic framework-bas...   
10399                                    Full Issue PDF.   
10767  Getting in touch with your senses: Mechanisms ...   
11329  Literature review of baseline information on n...   

                                                 journal  \
0                              Frontiers in pharmacology   
1                         Chemical biology & drug design   
2      Journal of Generic Medicines : Duplicate, mark...   
3                                   Structural chemistry   
4                                                Biology   
...                                                  ...   
9797                                Chemistry & industry   
9809                                       Nano research   
10399                               JACC. CardioOncology   
10767                        WIREs mechanisms of disease   
11329                       EFSA Supporting Publications   

                                                 authors  \
0      Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   
1      Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   
2                                                      0   
3      Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   
4                                    Mailem RC, Tayo LL.   
...                                                  ...   
9797                                                   0   
9809             Yang M, Zhang J, Wei Y, Zhang J, Tao C.   
10399                                                  0   
10767                                Gupta S, Butler SJ.   
11329  Dávalos A, Henriques R, Latasa M, Laparra M, C...   

                                 doi  \
0          10.3389/fphar.2022.974849   
1                 10.1111/cbdd.14136   
2                                  0   
3         10.1007/s11224-022-02020-z   
4            10.3390/biology11121827   
...                              ...   
9797            10.1002/cind.842_3.x   
9809       10.1007/s12274-022-4302-x   
10399  10.1016/s2666-0873(20)30180-0   
10767              10.1002/wsbm.1520   
11329                              0   

                                                 pdf_url  \
0      https://europepmc.org/articles/PMC9549161?pdf=...   
1      https://europepmc.org/articles/PMC9539342?pdf=...   
2      https://europepmc.org/articles/PMC9357751?pdf=...   
3      https://europepmc.org/articles/PMC9346052?pdf=...   
4      https://europepmc.org/articles/PMC9775208?pdf=...   
...                                                  ...   
9797   https://europepmc.org/articles/PMC7162151?pdf=...   
9809   https://europepmc.org/articles/PMC9094125?pdf=...   
10399  https://europepmc.org/articles/PMC7492056?pdf=...   
10767  https://europepmc.org/articles/PMC8459260?pdf=...   
11329  https://europepmc.org/articles/PMC7163523?pdf=...   

                                                    text  
0      Sir James Black, a winner of the 1988 No

In [ ]:
with open('2023-01-15_pmc_search_results_full_text_merged.pickle', 'wb') as f:
  pickle.dump(pmc_search_results_full_text_merged, f)

## 6. Check combined PMC DataFrame for missing data

Check that there are no null values in the combined DataFrame.

In [ ]:
pmc_search_results_full_text_merged.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11397 entries, 0 to 11329
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   pmcid      11397 non-null  object
 1   published  11397 non-null  object
 2   revised    11397 non-null  object
 3   title      11397 non-null  object
 4   journal    11397 non-null  object
 5   authors    11397 non-null  object
 6   doi        11397 non-null  object
 7   pdf_url    11397 non-null  object
 8   text       11397 non-null  object
dtypes: object(9)
memory usage: 890.4+ KB


Check using .isna() on the text column.

In [ ]:
pmc_search_results_full_text_merged[pmc_search_results_full_text_merged['text'].isna()]

Empty DataFrame
Columns: [pmcid, published, revised, title, journal, authors, doi, pdf_url, text]
Index: []

View a few of the 118 exceptions to see that they now have text in the combined DataFrame.

In [ ]:
pmc_search_results_full_text_merged[pmc_search_results_full_text_merged['pmcid']=='PMC9538661']

pmcid   published     revised  \
64  PMC9538661  2022-10-13  2022-11-22   

                                                title      journal  \
64  Recent Drug Development and Medicinal Chemistr...  ChemMedChem   

                                          authors                     doi  \
64  Ghosh AK, Mishevich JL, Mesecar A, Mitsuya H.  10.1002/cmdc.202200440   

                                              pdf_url  \
64  https://europepmc.org/articles/PMC9538661?pdf=...   

                                                 text  
64  \nSevere acute respiratory syndrome coronaviru...

In [ ]:
pmc_search_results_full_text_merged[pmc_search_results_full_text_merged['pmcid']=='PMC7163523']

pmcid   published     revised  \
11329  PMC7163523  2019-08-01  2021-02-16   

                                                   title  \
11329  Literature review of baseline information on n...   

                            journal  \
11329  EFSA Supporting Publications   

                                                 authors doi  \
11329  Dávalos A, Henriques R, Latasa M, Laparra M, C...   0   

                                                 pdf_url  \
11329  https://europepmc.org/articles/PMC7163523?pdf=...   

                                                    text  
11329  \nThis part provides baseline information on t...

In [ ]:
pmc_search_results_full_text_merged[pmc_search_results_full_text_merged['pmcid']=='PMC7640961']

pmcid   published     revised  \
2583  PMC7640961  2020-09-27  2020-12-18   

                                                  title  \
2583  Repurposing of FDA-Approved Toremifene to Trea...   

                           journal              authors  \
2583  Journal of proteome research  Martin WR, Cheng F.   

                                doi  \
2583  10.1021/acs.jproteome.0c00397   

                                                pdf_url  \
2583  https://europepmc.org/articles/PMC7640961?pdf=...   

                                                   text  
2583  \nAs of August 4, 2020, there are over 18 mill...

Check that there are no instances of '0' in the text column.

In [ ]:
pmc_search_results_full_text_merged[pmc_search_results_full_text_merged['text']=='0']

Empty DataFrame
Columns: [pmcid, published, revised, title, journal, authors, doi, pdf_url, text]
Index: []

Check that there are no instances of 'Abstract' in the title column.

In [ ]:
pmc_search_results_full_text_merged[pmc_search_results_full_text_merged['title'].str.contains('Abstract')]

Empty DataFrame
Columns: [pmcid, published, revised, title, journal, authors, doi, pdf_url, text]
Index: []


When we first ran the parse_pdf function for the PMC data and created the futures_map dictionary, there were 13 proxy exceptions with an empty string.

In [ ]:
with open('2023-01-06_europepmc_ft_xml.pickle', 'rb') as f:
    dl_results = pickle.load(f)

In [ ]:
dl_results.count('0')

13

In [ ]:
with open('2023-01-06_europepmc_df_json_ft_urls_13_exceptions.pickle', 'rb') as f:
    exceptions = pickle.load(f)

In [ ]:
exceptions

{'PMC8018918': Exception(''),
 'PMC7640961': Exception(''),
 'PMC8018905': Exception(''),
 'PMC7098069': Exception(''),
 'PMC7382535': Exception(''),
 'PMC7936759': Exception(''),
 'PMC8014535': Exception(''),
 'PMC7383733': Exception(''),
 'PMC7558230': Exception(''),
 'PMC8115429': Exception(''),
 'PMC7497212': Exception(''),
 'PMC7321661': Exception(''),
 'PMC8459260': Exception('')}

Check for a couple of the 13 exceptions that they now appear with text in the new combined DataFrame.

In [ ]:
test_exceptions = pmc_search_results_full_text_merged[pmc_search_results_full_text_merged.isin(['PMC8018918','PMC7558230']).any(axis=1)]
test_exceptions

pmcid   published     revised  \
2353  PMC8018918  2021-04-03  2022-11-21   
5714  PMC7558230  2020-10-15  2021-01-10   

                                                  title  \
2353  Virtual high throughput screening: Potential i...   
5714  Evaluation of mechanisms of action of re-purpo...   

                               journal  \
2353  European journal of pharmacology   
5714               Cellular immunology   

                                                authors  \
2353  Jade D, Ayyamperumal S, Tallapaneni V, Joghee ...   
5714  Rajaiah R, Abhilasha KV, Shekar MA, Vogel SN, ...   

                                doi  \
2353   10.1016/j.ejphar.2021.174082   
5714  10.1016/j.cellimm.2020.104240   

                                                pdf_url  \
2353  https://europepmc.org/articles/PMC8018918?pdf=...   
5714  https://europepmc.org/articles/PMC7558230?pdf=...   

                                                   text  
2353  \nIn December 2019, rare pneumonia, now called...  
5714  \nIn the past two decades, the world populatio...

Check for any more rows with an empty string in the text column.

In [ ]:
pmc_search_results_full_text_merged[pmc_search_results_full_text_merged['text']=='']

pmcid   published     revised  \
1772  PMC9364697  2022-07-06  2022-10-06   
1305  PMC8010379  2021-03-31  2022-11-08   

                                                  title  \
1772  3<sup>a</sup> edizione Giornate della ricerca ...   
1305  Repurposing antiviral drugs on recently emerge...   

                                         journal  \
1772  Journal of preventive medicine and hygiene   
1305                Materials today. Proceedings   

                                                authors  \
1772                                                  0   
1305  Swathi K, Nikitha B, Chandrakala B, Lakshmanad...   

                                     doi  \
1772  10.15167/2421-4248/jpmh2022.63.1s1   
1305         10.1016/j.matpr.2021.03.143   

                                                pdf_url text  
1772  https://europepmc.org/articles/PMC9364697?pdf=...       
1305  https://europepmc.org/articles/PMC8010379?pdf=...

The two rows that were identified earlier are there: PMC9364697 the Italian edition journal and PMC8010379 the withdrawn article.

In [ ]:
pmc_search_results_full_text_merged[pmc_search_results_full_text_merged['pmcid']=='PMC9364697']

pmcid   published     revised  \
1772  PMC9364697  2022-07-06  2022-10-06   

                                                  title  \
1772  3<sup>a</sup> edizione Giornate della ricerca ...   

                                         journal authors  \
1772  Journal of preventive medicine and hygiene       0   

                                     doi  \
1772  10.15167/2421-4248/jpmh2022.63.1s1   

                                                pdf_url text  
1772  https://europepmc.org/articles/PMC9364697?pdf=...

In [ ]:
pmc_search_results_full_text_merged[pmc_search_results_full_text_merged['pmcid']=='PMC8010379']

pmcid   published     revised  \
1305  PMC8010379  2021-03-31  2022-11-08   

                                                  title  \
1305  Repurposing antiviral drugs on recently emerge...   

                           journal  \
1305  Materials today. Proceedings   

                                                authors  \
1305  Swathi K, Nikitha B, Chandrakala B, Lakshmanad...   

                              doi  \
1305  10.1016/j.matpr.2021.03.143   

                                                pdf_url text  
1305  https://europepmc.org/articles/PMC8010379?pdf=...

Create a copy of the combined DataFrame and drop the two rows.

In [ ]:
pmc_search_results_full_text_merged_new = pmc_search_results_full_text_merged.copy()

In [ ]:
pmc_search_results_full_text_merged_new.drop([1772,1305], axis=0, inplace=True)
pmc_search_results_full_text_merged_new

pmcid   published     revised  \
0      PMC9549161  2022-09-26  2022-10-14   
1      PMC9539342  2022-09-22  2022-11-12   
2      PMC9357751  2022-12-01  2022-12-05   
3      PMC9346052  2022-08-03  2022-09-05   
4      PMC9775208  2022-12-15  2022-12-25   
...           ...         ...         ...   
9797   PMC7162151  2020-02-28  2020-04-21   
9809   PMC9094125  2022-05-11  2022-07-16   
10399  PMC7492056  2020-09-15  2020-09-28   
10767  PMC8459260  2021-02-25  2022-01-25   
11329  PMC7163523  2019-08-01  2021-02-16   

                                                   title  \
0           Drug repositioning: A bibliometric analysis.   
1      A review on computer-aided chemogenomics and d...   
2      Repurposing Molnupiravir as a new opportunity ...   
3      Scope of repurposed drugs against the potentia...   
4      Drug Repurposing Using Gene Co-Expression and ...   
...                                                  ...   
9797                                               News.   
9809   Recent advances in metal-organic framework-bas...   
10399                                    Full Issue PDF.   
10767  Getting in touch with your senses: Mechanisms ...   
11329  Literature review of baseline information on n...   

                                                 journal  \
0                              Frontiers in pharmacology   
1                         Chemical biology & drug design   
2      Journal of Generic Medicines : Duplicate, mark...   
3                                   Structural chemistry   
4                                                Biology   
...                                                  ...   
9797                                Chemistry & industry   
9809                                       Nano research   
10399                               JACC. CardioOncology   
10767                        WIREs mechanisms of disease   
11329                       EFSA Supporting Publications   

                                                 authors  \
0      Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   
1      Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   
2                                                      0   
3      Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   
4                                    Mailem RC, Tayo LL.   
...                                                  ...   
9797                                                   0   
9809             Yang M, Zhang J, Wei Y, Zhang J, Tao C.   
10399                                                  0   
10767                                Gupta S, Butler SJ.   
11329  Dávalos A, Henriques R, Latasa M, Laparra M, C...   

                                 doi  \
0          10.3389/fphar.2022.974849   
1                 10.1111/cbdd.14136   
2                                  0   
3         10.1007/s11224-022-02020-z   
4            10.3390/biology11121827   
...                              ...   
9797            10.1002/cind.842_3.x   
9809       10.1007/s12274-022-4302-x   
10399  10.1016/s2666-0873(20)30180-0   
10767              10.1002/wsbm.1520   
11329                              0   

                                                 pdf_url  \
0      https://europepmc.org/articles/PMC9549161?pdf=...   
1      https://europepmc.org/articles/PMC9539342?pdf=...   
2      https://europepmc.org/articles/PMC9357751?pdf=...   
3      https://europepmc.org/articles/PMC9346052?pdf=...   
4      https://europepmc.org/articles/PMC9775208?pdf=...   
...                                                  ...   
9797   https://europepmc.org/articles/PMC7162151?pdf=...   
9809   https://europepmc.org/articles/PMC9094125?pdf=...   
10399  https://europepmc.org/articles/PMC7492056?pdf=...   
10767  https://europepmc.org/articles/PMC8459260?pdf=...   
11329  https://europepmc.org/articles/PMC7163523?pdf=...   

                                                    text  
0      Sir James Black, a winner of the 1988 No

In [ ]:
len(pmc_search_results_full_text_merged_new)

11395

Reset index

In [ ]:
pmc_search_results_full_text_merged_new.reset_index(drop=True, inplace=True)

Check for one of the articles (row 1772	PMC9364697) that it has been dropped and the DataFrame reindexed.

In [ ]:
pmc_search_results_full_text_merged_new.loc[1771:1773]

pmcid   published     revised  \
1771  PMC8803929  2022-02-01  2022-11-07   
1772  PMC8889327  2022-02-01  2022-03-11   
1773  PMC8929333  2022-02-01  2022-03-25   

                                                  title  \
1771  CoVac501, a self-adjuvanting peptide vaccine c...   
1772             Organoid Studies in COVID-19 Research.   
1773  The interacting physiology of COVID-19 and the...   

                                   journal  \
1771                        Cell discovery   
1772   International journal of stem cells   
1773  Pharmacology research & perspectives   

                                                authors  \
1771  Long Y, Sun J, Song TZ, Liu T, Tang F, Zhang X...   
1772                          Kim J, Koo BK, Clevers H.   
1773  Lumbers ER, Head R, Smith GR, Delforce SJ, Jar...   

                             doi  \
1771  10.1038/s41421-021-00370-2   
1772          10.15283/ijsc21251   
1773            10.1002/prp2.917   

                                                pdf_url  \
1771  https://europepmc.org/articles/PMC8803929?pdf=...   
1772  https://europepmc.org/articles/PMC8889327?pdf=...   
1773  https://europepmc.org/articles/PMC8929333?pdf=...   

                                                   text  
1771  The uncontrolled transmission and ongoing evol...  
1772  Global pandemics, such as bubonic plague and S...  
1773  ACE2angiotensin‐converting enzyme 2ACEIsangiot...

In [ ]:
with open('2023-01-15_pmc_search_results_full_text_merged_new.pickle', 'wb') as f:
  pickle.dump(pmc_search_results_full_text_merged_new, f)

In [ ]:
len(pmc_search_results_full_text_merged_new)

11395

Rename pmcid column heading in PMC DataFrame to article-id before combining with arXiv DataFrame.

In [ ]:
pmc_search_results_full_text_merged_article_id = pmc_search_results_full_text_merged_new.copy()

In [ ]:
pmc_search_results_full_text_merged_article_id.rename({'pmcid': 'article_id'}, axis=1, inplace=True)
pmc_search_results_full_text_merged_article_id

article_id   published     revised  \
0      PMC9549161  2022-09-26  2022-10-14   
1      PMC9539342  2022-09-22  2022-11-12   
2      PMC9357751  2022-12-01  2022-12-05   
3      PMC9346052  2022-08-03  2022-09-05   
4      PMC9775208  2022-12-15  2022-12-25   
...           ...         ...         ...   
11390  PMC7162151  2020-02-28  2020-04-21   
11391  PMC9094125  2022-05-11  2022-07-16   
11392  PMC7492056  2020-09-15  2020-09-28   
11393  PMC8459260  2021-02-25  2022-01-25   
11394  PMC7163523  2019-08-01  2021-02-16   

                                                   title  \
0           Drug repositioning: A bibliometric analysis.   
1      A review on computer-aided chemogenomics and d...   
2      Repurposing Molnupiravir as a new opportunity ...   
3      Scope of repurposed drugs against the potentia...   
4      Drug Repurposing Using Gene Co-Expression and ...   
...                                                  ...   
11390                                              News.   
11391  Recent advances in metal-organic framework-bas...   
11392                                    Full Issue PDF.   
11393  Getting in touch with your senses: Mechanisms ...   
11394  Literature review of baseline information on n...   

                                                 journal  \
0                              Frontiers in pharmacology   
1                         Chemical biology & drug design   
2      Journal of Generic Medicines : Duplicate, mark...   
3                                   Structural chemistry   
4                                                Biology   
...                                                  ...   
11390                               Chemistry & industry   
11391                                      Nano research   
11392                               JACC. CardioOncology   
11393                        WIREs mechanisms of disease   
11394                       EFSA Supporting Publications   

                                                 authors  \
0      Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   
1      Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   
2                                                      0   
3      Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   
4                                    Mailem RC, Tayo LL.   
...                                                  ...   
11390                                                  0   
11391            Yang M, Zhang J, Wei Y, Zhang J, Tao C.   
11392                                                  0   
11393                                Gupta S, Butler SJ.   
11394  Dávalos A, Henriques R, Latasa M, Laparra M, C...   

                                 doi  \
0          10.3389/fphar.2022.974849   
1                 10.1111/cbdd.14136   
2                                  0   
3         10.1007/s11224-022-02020-z   
4            10.3390/biology11121827   
...                              ...   
11390           10.1002/cind.842_3.x   
11391      10.1007/s12274-022-4302-x   
11392  10.1016/s2666-0873(20)30180-0   
11393              10.1002/wsbm.1520   
11394                              0   

                                                 pdf_url  \
0      https://europepmc.org/articles/PMC9549161?pdf=...   
1      https://europepmc.org/articles/PMC9539342?pdf=...   
2      https://europepmc.org/articles/PMC9357751?pdf=...   
3      https://europepmc.org/articles/PMC9346052?pdf=...   
4      https://europepmc.org/articles/PMC9775208?pdf=...   
...                                                  ...   
11390  https://europepmc.org/articles/PMC7162151?pdf=...   
11391  https://europepmc.org/articles/PMC9094125?pdf=...   
11392  https://europepmc.org/articles/PMC7492056?pdf=...   
11393  https://europepmc.org/articles/PMC8459260?pdf=...   
11394  https://europepmc.org/articles/PMC7163523?pdf=...   

                                                    text  
0      Sir James Black, a winner of the 19

## 7. Import arXiv data

Import DataFrame for 301 articles with full text, and rows for three unavailable PDFs dropped.

In [ ]:
with open('2023-01-06_article_results_arxiv_301_full_text.pickle', 'rb') as f:
    article_results_arxiv_301 = pickle.load(f)

In [ ]:
article_results_arxiv_301

arxiv-id   published     revised  \
0    2109.06377v4  2021-09-14  2022-12-22   
1    2212.09867v1  2022-12-19  2022-12-19   
2    2212.09610v1  2022-12-19  2022-12-19   
3    2212.03911v1  2022-12-07  2022-12-07   
4    2212.01575v1  2022-12-03  2022-12-03   
..            ...         ...         ...   
296  2003.13665v1  2020-03-30  2020-03-30   
297  2003.14258v1  2020-03-30  2020-03-30   
298  2003.12454v1  2020-03-26  2020-03-26   
299  2003.04524v1  2020-03-10  2020-03-10   
300  2003.00751v1  2020-03-02  2020-03-02   

                                                 title journal  \
0    ASGARD: A Single-cell Guided pipeline to Aid R...    None   
1    Detecting Contradictory COVID-19 Drug Efficacy...    None   
2    Drying of Bio-colloidal Sessile Droplets: Adva...    None   
3    Analysis of Drug repurposing Knowledge graphs ...    None   
4    Multi-view deep learning based molecule design...    None   
..                                                 ...     ...   
296  Genomics-guided molecular maps of coronavirus ...    None   
297  Nanomechanical sonification of the 2019-nCoV c...    None   
298  A Machine Learning alternative to placebo-cont...    None   
299  Old Drugs for Newly Emerging Viral Disease, CO...    None   
300  Regulation of interferon production as a poten...    None   

                                               authors   doi  \
0    Bing He, Yao Xiao, Haodong Liang, Qianhui Huan...  None   
1    Daniel N. Sosa, Malavika Suresh, Christopher P...  None   
2           Anusuya Pal, Amalesh Gope, Anupam Sengupta  None   
3                                  Ajay Kumar Gogineni  None   
4    Chao Pang, Yu Wang, Yi Jiang, Ruheng Wang, Ran...  None   
..                                                 ...   ...   
296                                    Gennadi Glinsky  None   
297                                  Markus J. Buehler  None   
298   Ezequiel Alvarez, Federico Lamagna, Manuel Szewc  None   
299                                Mohammad Reza Dayer  None   
300             Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei  None   

                                      pdf_url  \
0    http://export.arxiv.org/pdf/2109.06377v4   
1    http://export.arxiv.org/pdf/2212.09867v1   
2    http://export.arxiv.org/pdf/2212.09610v1   
3    http://export.arxiv.org/pdf/2212.03911v1   
4    http://export.arxiv.org/pdf/2212.01575v1   
..                                        ...   
296  http://export.arxiv.org/pdf/2003.13665v1   
297  http://export.arxiv.org/pdf/2003.14258v1   
298  http://export.arxiv.org/pdf/2003.12454v1   
299  http://export.arxiv.org/pdf/2003.04524v1   
300  http://export.arxiv.org/pdf/2003.00751v1   

                                                  text  
0    \nHeterogeneity, or more specifically, the div...  
1    \nThe COVID-19 pandemic caused by the novel SA...  
2    \nVirus Emulating Particles (VEP) White Blood ...  
3    \nIn the initial stages of a viral outbreak su...  
4    \nDe novo drug design is a time-consuming and ...  
..                                                 ...  
296  \nCoronavirus pandemic 2020 caused by the newl...  
297  \nProteins are the building blocks of virtuall...  
298  \nCurrent and last decades research in drug di...  
299  \nThe outbreak of coronavirus  in 2019-2020 is...  
300  \nNovel Coronavirus SARS-CoV-2 is a single str...  

[301 rows x 9 columns]

There is one outstanding exception where GROBID was unable to parse the PDF with Exception('[GENERAL] An exception occurred while running Grobid.').

In [ ]:
article_results_arxiv_301[article_results_arxiv_301['arxiv-id']=='2007.09186v3']

arxiv-id   published     revised  \
173  2007.09186v3  2020-07-17  2020-10-07   

                                                 title journal  \
173  AWS CORD-19 Search: A Neural Search Engine for...    None   

                                               authors   doi  \
173  Parminder Bhatia, Lan Liu, Kristjan Arumae, Ni...  None   

                                      pdf_url  text  
173  http://export.arxiv.org/pdf/2007.09186v3  None

Rename arxiv-id column heading in arXiv DataFrame to article_id before combining with PMC DataFrame.

In [ ]:
article_results_arxiv_301_article_id = article_results_arxiv_301.copy()

In [ ]:
article_results_arxiv_301_article_id.rename({'arxiv-id': 'article_id'}, axis=1, inplace=True)
article_results_arxiv_301_article_id

article_id   published     revised  \
0    2109.06377v4  2021-09-14  2022-12-22   
1    2212.09867v1  2022-12-19  2022-12-19   
2    2212.09610v1  2022-12-19  2022-12-19   
3    2212.03911v1  2022-12-07  2022-12-07   
4    2212.01575v1  2022-12-03  2022-12-03   
..            ...         ...         ...   
296  2003.13665v1  2020-03-30  2020-03-30   
297  2003.14258v1  2020-03-30  2020-03-30   
298  2003.12454v1  2020-03-26  2020-03-26   
299  2003.04524v1  2020-03-10  2020-03-10   
300  2003.00751v1  2020-03-02  2020-03-02   

                                                 title journal  \
0    ASGARD: A Single-cell Guided pipeline to Aid R...    None   
1    Detecting Contradictory COVID-19 Drug Efficacy...    None   
2    Drying of Bio-colloidal Sessile Droplets: Adva...    None   
3    Analysis of Drug repurposing Knowledge graphs ...    None   
4    Multi-view deep learning based molecule design...    None   
..                                                 ...     ...   
296  Genomics-guided molecular maps of coronavirus ...    None   
297  Nanomechanical sonification of the 2019-nCoV c...    None   
298  A Machine Learning alternative to placebo-cont...    None   
299  Old Drugs for Newly Emerging Viral Disease, CO...    None   
300  Regulation of interferon production as a poten...    None   

                                               authors   doi  \
0    Bing He, Yao Xiao, Haodong Liang, Qianhui Huan...  None   
1    Daniel N. Sosa, Malavika Suresh, Christopher P...  None   
2           Anusuya Pal, Amalesh Gope, Anupam Sengupta  None   
3                                  Ajay Kumar Gogineni  None   
4    Chao Pang, Yu Wang, Yi Jiang, Ruheng Wang, Ran...  None   
..                                                 ...   ...   
296                                    Gennadi Glinsky  None   
297                                  Markus J. Buehler  None   
298   Ezequiel Alvarez, Federico Lamagna, Manuel Szewc  None   
299                                Mohammad Reza Dayer  None   
300             Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei  None   

                                      pdf_url  \
0    http://export.arxiv.org/pdf/2109.06377v4   
1    http://export.arxiv.org/pdf/2212.09867v1   
2    http://export.arxiv.org/pdf/2212.09610v1   
3    http://export.arxiv.org/pdf/2212.03911v1   
4    http://export.arxiv.org/pdf/2212.01575v1   
..                                        ...   
296  http://export.arxiv.org/pdf/2003.13665v1   
297  http://export.arxiv.org/pdf/2003.14258v1   
298  http://export.arxiv.org/pdf/2003.12454v1   
299  http://export.arxiv.org/pdf/2003.04524v1   
300  http://export.arxiv.org/pdf/2003.00751v1   

                                                  text  
0    \nHeterogeneity, or more specifically, the div...  
1    \nThe COVID-19 pandemic caused by the novel SA...  
2    \nVirus Emulating Particles (VEP) White Blood ...  
3    \nIn the initial stages of a viral outbreak su...  
4    \nDe novo drug design is a time-consuming and ...  
..                                                 ...  
296  \nCoronavirus pandemic 2020 caused by the newl...  
297  \nProteins are the building blocks of virtuall...  
298  \nCurrent and last decades research in drug di...  
299  \nThe outbreak of coronavirus  in 2019-2020 is...  
300  \nNovel Coronavirus SARS-CoV-2 is a single str...  

[301 rows x 9 columns]

## 8. Combine PMC and arXiv data

Create a new merged dataset by concatenating the PMC and arXiv DataFrames with generic article-id column.

In [ ]:
pmc_arxiv_full_text_merged = pd.concat([pmc_search_results_full_text_merged_article_id, article_results_arxiv_301_article_id], ignore_index=True)

In [ ]:
pmc_arxiv_full_text_merged

article_id   published     revised  \
0        PMC9549161  2022-09-26  2022-10-14   
1        PMC9539342  2022-09-22  2022-11-12   
2        PMC9357751  2022-12-01  2022-12-05   
3        PMC9346052  2022-08-03  2022-09-05   
4        PMC9775208  2022-12-15  2022-12-25   
...             ...         ...         ...   
11691  2003.13665v1  2020-03-30  2020-03-30   
11692  2003.14258v1  2020-03-30  2020-03-30   
11693  2003.12454v1  2020-03-26  2020-03-26   
11694  2003.04524v1  2020-03-10  2020-03-10   
11695  2003.00751v1  2020-03-02  2020-03-02   

                                                   title  \
0           Drug repositioning: A bibliometric analysis.   
1      A review on computer-aided chemogenomics and d...   
2      Repurposing Molnupiravir as a new opportunity ...   
3      Scope of repurposed drugs against the potentia...   
4      Drug Repurposing Using Gene Co-Expression and ...   
...                                                  ...   
11691  Genomics-guided molecular maps of coronavirus ...   
11692  Nanomechanical sonification of the 2019-nCoV c...   
11693  A Machine Learning alternative to placebo-cont...   
11694  Old Drugs for Newly Emerging Viral Disease, CO...   
11695  Regulation of interferon production as a poten...   

                                                 journal  \
0                              Frontiers in pharmacology   
1                         Chemical biology & drug design   
2      Journal of Generic Medicines : Duplicate, mark...   
3                                   Structural chemistry   
4                                                Biology   
...                                                  ...   
11691                                               None   
11692                                               None   
11693                                               None   
11694                                               None   
11695                                               None   

                                                 authors  \
0      Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   
1      Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   
2                                                      0   
3      Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   
4                                    Mailem RC, Tayo LL.   
...                                                  ...   
11691                                    Gennadi Glinsky   
11692                                  Markus J. Buehler   
11693   Ezequiel Alvarez, Federico Lamagna, Manuel Szewc   
11694                                Mohammad Reza Dayer   
11695             Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei   

                              doi  \
0       10.3389/fphar.2022.974849   
1              10.1111/cbdd.14136   
2                               0   
3      10.1007/s11224-022-02020-z   
4         10.3390/biology11121827   
...                           ...   
11691                        None   
11692                        None   
11693                        None   
11694                        None   
11695                        None   

                                                 pdf_url  \
0      https://europepmc.org/articles/PMC9549161?pdf=...   
1      https://europepmc.org/articles/PMC9539342?pdf=...   
2      https://europepmc.org/articles/PMC9357751?pdf=...   
3      https://europepmc.org/articles/PMC9346052?pdf=...   
4      https://europepmc.org/articles/PMC9775208?pdf=...   
...                                                  ...   
11691           http://export.arxiv.org/pdf/2003.13665v1   
11692           http://export.arxiv.org/pdf/2003.14258v1   
11693           http://export.arxiv.org/pdf/2003.12454v1   
11694           http://export.arxiv.org/pdf/2003.04524v1   
11695           http://export.arxiv.org/pdf/2003.00751v1   

                                                    text  
0      Sir James Black, a winner of the 1988 Nobel Pr...

## 9. Check for duplicate titles

Check for duplicate titles, for example any titles appearing in both PMC and arXiv databases.

In [ ]:
duplicate_titles = pmc_arxiv_full_text_merged[pmc_arxiv_full_text_merged.duplicated('title', keep=False)].sort_values('title')
duplicate_titles

article_id   published     revised  \
3964     PMC9040466  2022-01-01  2022-12-01   
4449     PMC8596203  2021-11-17  2022-11-30   
11369    PMC7247787  2020-05-25  2020-09-28   
11392    PMC7492056  2020-09-15  2020-09-28   
5528     PMC7232076  2020-05-18  2021-01-27   
11677  2004.08491v1  2020-04-18  2020-04-18   
11353    PMC7280571  2020-05-22  2020-06-15   
11367    PMC7280669  2020-05-22  2020-06-15   
11390    PMC7162151  2020-02-28  2020-04-21   
8729     PMC7489339  2020-09-14  2021-06-29   
9000     PMC7218391  2020-05-13  2021-06-24   

                                                   title  \
3964   CKD: The burden of disease invisible to resear...   
4449   CKD: The burden of disease invisible to resear...   
11369                                    Full Issue PDF.   
11392                                    Full Issue PDF.   
5528   Is the anti-filarial drug diethylcarbamazine u...   
11677  Is the anti-filarial drug diethylcarbamazine u...   
11353                                              News.   
11367                                              News.   
11390                                              News.   
8729   Targeting the Linear Ubiquitin Assembly Comple...   
9000   Targeting the Linear Ubiquitin Assembly Comple...   

                                                 journal  \
3964                                          Nefrologia   
4449   Nefrologia : publicacion oficial de la Socieda...   
11369               JACC. Basic to translational science   
11392                               JACC. CardioOncology   
5528                                  Medical hypotheses   
11677                                               None   
11353                               Chemistry & industry   
11367                               Chemistry & industry   
11390                               Chemistry & industry   
8729                        Archivos de bronconeumologia   
9000                        Archivos de bronconeumologia   

                                                 authors  \
3964   AIRG-E, EKPF, ALCER, FRIAT, REDINREN, RICORS20...   
4449   AIRG-E, EKPF, ALCER, FRIAT, REDINREN, RICORS20...   
11369                                                  0   
11392                                                  0   
5528                     Abeygunasekera A, Jayasinghe S.   
11677         Anuruddha Abeygunasekera, Saroj Jayasinghe   
11353                                                  0   
11367                                                  0   
11390                                                  0   
8729                             Brazee PL, Sznajder JI.   
9000                             Brazee PL, Sznajder JI.   

                                 doi  \
3964    10.1016/j.nefroe.2021.09.005   
4449     10.1016/j.nefro.2021.09.004   
11369  10.1016/s2452-302x(20)30205-9   
11392  10.1016/s2666-0873(20)30180-0   
5528      10.1016/j.mehy.2020.109843   
11677                           None   
11353           10.1002/cind.3_845.x   
11367           10.1002/cind.5_845.x   
11390           10.1002/cind.842_3.x   
8729      10.1016/j.arbr.2020.04.008   
9000    10.1016/j.arbres.2020.04.019   

                                                 pdf_url  \
3964   https://europepmc.org/articles/PMC9040466?pdf=...   
4449   https://europepmc.org/articles/PMC8596203?pdf=...   
11369  https://europepmc.org/articles/PMC7247787?pdf=...   
11392  https://europepmc.org/articles/PMC7492056?pdf=...   
5528   https://europepmc.org/articles/PMC7232076?pdf=...   
11677           http://export.arxiv.org/pdf/2004.08491v1   
11353  https://europepmc.org/articles/PMC7280571?pdf=...   
11367  https://europepmc.org/articles/PMC7280669?pdf=...   
11390  https://europepmc.org/articles/PMC7162151?pdf=...   
8729   https://europepmc.org/articles/PMC7489339?pdf=...   
9000   https://europepmc.org/articles/PMC7218391?pdf=...   

                                                    text  
3964   The present m

In [ ]:
len(duplicate_titles)

11

There are 11 duplicate titles requiring further investigation.

*   An English and Spanish edition appear in PMC - we will drop the latter (PMC8596203).
*   Two entitled 'Full Issue PDF' in PMC are different documents - both will be kept.
*   One title appears in PMC and arXiv - we will keep the PMC version with journal reference, DOI and more recent publish and revised dates and drop the latter (2004.08491v1).
*   Three from the same journal are entitled 'News' in PMC but are different articles - we will keep all.
*   Another English and Spanish edition appear in PMC - we will drop the latter (PMC7218391).




In [ ]:
rows_to_drop = pmc_arxiv_full_text_merged[pmc_arxiv_full_text_merged.isin(['PMC8596203','2004.08491v1', 'PMC7218391']).any(axis=1)]
rows_to_drop

article_id   published     revised  \
4449     PMC8596203  2021-11-17  2022-11-30   
9000     PMC7218391  2020-05-13  2021-06-24   
11677  2004.08491v1  2020-04-18  2020-04-18   

                                                   title  \
4449   CKD: The burden of disease invisible to resear...   
9000   Targeting the Linear Ubiquitin Assembly Comple...   
11677  Is the anti-filarial drug diethylcarbamazine u...   

                                                 journal  \
4449   Nefrologia : publicacion oficial de la Socieda...   
9000                        Archivos de bronconeumologia   
11677                                               None   

                                                 authors  \
4449   AIRG-E, EKPF, ALCER, FRIAT, REDINREN, RICORS20...   
9000                             Brazee PL, Sznajder JI.   
11677         Anuruddha Abeygunasekera, Saroj Jayasinghe   

                                doi  \
4449    10.1016/j.nefro.2021.09.004   
9000   10.1016/j.arbres.2020.04.019   
11677                          None   

                                                 pdf_url  \
4449   https://europepmc.org/articles/PMC8596203?pdf=...   
9000   https://europepmc.org/articles/PMC7218391?pdf=...   
11677           http://export.arxiv.org/pdf/2004.08491v1   

                                                    text  
4449   The present manuscript summarizes key features...  
9000   Seasonal influenza A viral infection affects a...  
11677  \nSARS-CoV-2 virus has caused a pandemic with ...

Create a new combined DataFrame dropping the three rows.

In [ ]:
pmc_arxiv_full_text_merged_new = pd.concat([pmc_arxiv_full_text_merged, rows_to_drop]).drop_duplicates(keep=False).reset_index(drop=True)
pmc_arxiv_full_text_merged_new

article_id   published     revised  \
0        PMC9549161  2022-09-26  2022-10-14   
1        PMC9539342  2022-09-22  2022-11-12   
2        PMC9357751  2022-12-01  2022-12-05   
3        PMC9346052  2022-08-03  2022-09-05   
4        PMC9775208  2022-12-15  2022-12-25   
...             ...         ...         ...   
11688  2003.13665v1  2020-03-30  2020-03-30   
11689  2003.14258v1  2020-03-30  2020-03-30   
11690  2003.12454v1  2020-03-26  2020-03-26   
11691  2003.04524v1  2020-03-10  2020-03-10   
11692  2003.00751v1  2020-03-02  2020-03-02   

                                                   title  \
0           Drug repositioning: A bibliometric analysis.   
1      A review on computer-aided chemogenomics and d...   
2      Repurposing Molnupiravir as a new opportunity ...   
3      Scope of repurposed drugs against the potentia...   
4      Drug Repurposing Using Gene Co-Expression and ...   
...                                                  ...   
11688  Genomics-guided molecular maps of coronavirus ...   
11689  Nanomechanical sonification of the 2019-nCoV c...   
11690  A Machine Learning alternative to placebo-cont...   
11691  Old Drugs for Newly Emerging Viral Disease, CO...   
11692  Regulation of interferon production as a poten...   

                                                 journal  \
0                              Frontiers in pharmacology   
1                         Chemical biology & drug design   
2      Journal of Generic Medicines : Duplicate, mark...   
3                                   Structural chemistry   
4                                                Biology   
...                                                  ...   
11688                                               None   
11689                                               None   
11690                                               None   
11691                                               None   
11692                                               None   

                                                 authors  \
0      Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   
1      Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   
2                                                      0   
3      Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   
4                                    Mailem RC, Tayo LL.   
...                                                  ...   
11688                                    Gennadi Glinsky   
11689                                  Markus J. Buehler   
11690   Ezequiel Alvarez, Federico Lamagna, Manuel Szewc   
11691                                Mohammad Reza Dayer   
11692             Xiaobing Deng, Xiaoyu Yu, Jianfeng Pei   

                              doi  \
0       10.3389/fphar.2022.974849   
1              10.1111/cbdd.14136   
2                               0   
3      10.1007/s11224-022-02020-z   
4         10.3390/biology11121827   
...                           ...   
11688                        None   
11689                        None   
11690                        None   
11691                        None   
11692                        None   

                                                 pdf_url  \
0      https://europepmc.org/articles/PMC9549161?pdf=...   
1      https://europepmc.org/articles/PMC9539342?pdf=...   
2      https://europepmc.org/articles/PMC9357751?pdf=...   
3      https://europepmc.org/articles/PMC9346052?pdf=...   
4      https://europepmc.org/articles/PMC9775208?pdf=...   
...                                                  ...   
11688           http://export.arxiv.org/pdf/2003.13665v1   
11689           http://export.arxiv.org/pdf/2003.14258v1   
11690           http://export.arxiv.org/pdf/2003.12454v1   
11691           http://export.arxiv.org/pdf/2003.04524v1   
11692           http://export.arxiv.org/pdf/2003.00751v1   

                                                    text  
0      Sir James Black, a winner of the 1988 Nobel Pr...

Check for one of the articles (row 11677 2004.08491v1) that it has been dropped and the DataFrame reindexed.

In [ ]:
pmc_arxiv_full_text_merged_new.loc[11676:11678]

article_id   published     revised  \
11676  2004.07750v1  2020-04-16  2020-04-16   
11677  2003.10642v2  2020-03-24  2020-04-15   
11678  2004.07086v1  2020-04-15  2020-04-15   

                                                   title  \
11676  Extracting the effective contact rate of COVID...   
11677  In Silico Investigations on the Potential Inhi...   
11678  Prediction of potential inhibitors for RNA-dep...   

                                                 journal  \
11676                                               None   
11677                                               None   
11678  International Journal of Biological Macromolec...   

                                                 authors  \
11676       Gaurav Goswami, Jayanti Prasad, Mansi Dhuria   
11677  Ambrish Kumar Srivastava, Abhishek Kumar, Garg...   
11678  Md. Sorwer Alam Parvez, Md. Adnan Karim, Mahmu...   

                                  doi  \
11676                            None   
11677                            None   
11678  10.1016/j.ijbiomac.2020.09.098   

                                        pdf_url  \
11676  http://export.arxiv.org/pdf/2004.07750v1   
11677  http://export.arxiv.org/pdf/2003.10642v2   
11678  http://export.arxiv.org/pdf/2004.07086v1   

                                                    text  
11676  \nWithin a few months of its first outbreak, C...  
11677  \nAt the beginning of this year, the coronavir...  
11678  \nThe pandemic Corona Virus Disease 19 (COVID-...

In [ ]:
len(pmc_arxiv_full_text_merged_new)

11693

In [ ]:
with open('2023-01-15_pmc_arxiv_full_text_merged_new.pickle', 'wb') as f:
  pickle.dump(pmc_arxiv_full_text_merged_new, f)

This just leaves the one arXiv exception GROBID did not extract text for which will be handled separately using more manual methods.

In [ ]:
pmc_arxiv_full_text_merged_new.loc[pmc_arxiv_full_text_merged_new['text'].isna()]

article_id   published     revised  \
11566  2007.09186v3  2020-07-17  2020-10-07   

                                                   title journal  \
11566  AWS CORD-19 Search: A Neural Search Engine for...    None   

                                                 authors   doi  \
11566  Parminder Bhatia, Lan Liu, Kristjan Arumae, Ni...  None   

                                        pdf_url  text  
11566  http://export.arxiv.org/pdf/2007.09186v3  None